test/js/node/net/node-net-server.test.ts

```diff
@@ -282,9 +282,7 @@ describe("net.createServer listen", () => {
         } catch (e) {
           err = e as Error;
         }
-        expect(err).not.toBeNull();
-        expect(err!.message).toBe("Failed to connect");
-        expect(err!.name).toBe("ECONNREFUSED");
+        expect(err).toBeNull();
```

Comment:
If the server is listening on `0.0.0.0`, and the client connects using `::`, should this be successful?


node.js server and client

```JavaScript
const net = require("net");

const server = net.createServer(socket => {
  console.log("Client connected");

  socket.on("data", data => {
    console.log(`Received data from client: ${data}`);

    socket.write(data);
  });

  socket.on("end", () => {
    console.log("Client disconnected");
  });
});

server.listen(8000, "0.0.0.0", () => {
  console.log("Server listening on", server.address());
});

const client = net.createConnection(
  {
    host: "::",
    // host: "127.0.0.1",
    port: 8000,
  },
  () => {
    console.log("Connected to server successfully.");

    client.write("Hello, server!");

    client.end();
  },
);

client.on("data", data => {
  console.log(`Received data from server: ${data}`);
});

client.on("end", () => {
  console.log("Disconnected from server");
});

client.on("error", err => {
  console.error(`Error connecting to server: ${err.message}`);
});

```

Python client

```Python
import socket

server_address = ("::", 8000)
# server_address = ("0.0.0.0", 8000)

with socket.create_connection(server_address) as client:
    print(f"Connected to server {server_address}")

    message = "Hello, server!"
    client.sendall(message.encode("utf-8"))
    print(f"Sent data to server: {message}")

    data = client.recv(1024)
    print(f'Received data from server: {data.decode("utf-8")}')

```